refinery/data_set_manager/views.py

```diff
@@ -1059,11 +1060,20 @@ def post(self, request):
             return HttpResponseForbidden()
 
         logger.debug("Adding files to data set '%s'", data_set)
-        for file_store_item in data_set.get_file_store_items():
-            if (file_store_item.source.startswith(
+        node = get_object_or_404(Node, uuid=request.data.get('node_uuid'))
```

Comment:
If `node_uuid` is not a part of the request there will be a 404 returned.
If the submitted `node_uuid` does not correspond to the `data_set`, then this would still go ahead and try to run a `import_file` task for said `Node`. I can add error handling for that case.

Reply:
Shouldn't that be a 400 in case of a missing `node_uuid`?

Yes, Node permission check is necessary to avoid a potential security hole and in that case supplying a `data_set_uuid` should be redundant.

Reply:
Since a request without a `node_uuid` is invalid it should be a 400. Actually, that was the original implementation: https://github.com/refinery-platform/refinery-platform/blob/ab9eb18b2a847d5e347555ae19b895168f3fcc08/refinery/data_set_manager/views.py#L1002